packages/react-vapor/src/components/limit/Limit.tsx

```diff
@@ -0,0 +1,210 @@
+import {IDispatch} from '../../utils/ReduxUtils';
+import {changeInputValue} from '../input/InputActions';
+import {InputSelectors} from '../input/InputSelectors';
+
+export interface LimitOwnProps {
+    id?: string;
+    title: string;
+    limitLabel?: string;
+    usage?: number;
+    limit?: number;
+    isLimitEditable?: boolean;
+    isLimitTheGoalToReach?: boolean;
+    isHistoryIncluded?: boolean;
+}
+
+const mapStateToProps = (state: IReactVaporState, ownProps: LimitOwnProps) => {
+    const ownLimit: string = ownProps.limit.toString() || Limit.defaultProps.limit.toString();
+    const currentLimit: string = InputSelectors.getValue(state, {id: ownProps.id}) || ownLimit;
+    return {
+        currentLimit,
+    };
+};
+
+const mapDispatchToProps = (dispatch: IDispatch, ownProps: LimitOwnProps) => ({
+    onChangeLimit: (limitValue: string) => {
+        dispatch(changeInputValue(ownProps.id, limitValue, true));
+    },
+});
+
+export interface LimitProps
+    extends LimitOwnProps,
+        ReturnType<typeof mapDispatchToProps>,
+        ReturnType<typeof mapStateToProps> {}
+
+export const LimitDisconnect: React.FunctionComponent<LimitProps> = ({
+    id,
+    title,
+    limitLabel,
+    usage,
+    limit,
+    currentLimit,
+    isLimitEditable,
+    isLimitTheGoalToReach,
+    isHistoryIncluded,
+    onChangeLimit: onChangeLimit,
+}) => {
+    return (
+        <div className="limit-box mb2">
+            <div className="limit-box-main p2 pb1">
+                <HeaderDivision limitTitle={title} isHistory={isHistoryIncluded} />
+                <ContentDivision
+                    id={id}
+                    usageValue={usage}
+                    limitValue={limit}
+                    isEditable={isLimitEditable}
+                    limitTitle={limitLabel}
+                    onChangeLimit={onChangeLimit}
+                />
+            </div>
+            <ProgressBar usageValue={usage} isLimitTheGoal={isLimitTheGoalToReach} limitValue={+currentLimit} />
+        </div>
+    );
+};
+
+const HeaderDivision = ({limitTitle, isHistory}: {limitTitle: string; isHistory: boolean}) => {
+    return (
+        <div className="flex space-between">
+            <label className="form-control-label"> {limitTitle}</label>
+            <HistoryIcon isHistory={isHistory} />
+        </div>
+    );
+};
+
+const HistoryIcon = ({isHistory}: {isHistory: boolean}) => {
+    return (
+        isHistory && (
+            <span className="icon mod-lg limit-history-button">
+                <Svg svgName={VaporSVG.svg.menuAnalytics.name} className="fill-medium-blue" />
+            </span>
+        )
+    );
+};
+
+const ContentDivision = ({
+    id,
+    usageValue,
+    limitValue,
+    isEditable,
+    limitTitle,
+    onChangeLimit,
+}: {
+    id: string;
+    usageValue: number;
+    limitValue: number;
+    isEditable: boolean;
+    limitTitle: string;
+    onChangeLimit: (limit: string) => void;
```

Comment:
I would do an interface here in order to extract the logic. 
But even better, in order to reuse logic. Since the props you are passing here are mostly the same than the `LimitDisconnect` props you could create a type by picking the props from the `LimitProps` like this:
`type NewType = Pick<LimitProps, 'Id', 'usage' [...so on] >`  But of course you would have to keep the name of your props the same and not modify them like you did (ex: limit => limitValue).

Im not 100% sure if you can pick type properties since they came from a `returnType<typeof mapStateToProps>` but this sure worth the shot to try.

worst case extract all those properties in a distinct interface.

Reply:
Thanks for this tip! Actually, would it be better that I set all my props with Pick? For instance, if I set my ContentProps this way: 
`type ContentProps = type ContentProps = Pick<LimitProps, 'id' | 'usage' | 'limit' | 'isLimitEditable' | 'limitLabel' | 'onChangeLimit'>; `
Then I set my UsageProps this way:
`type UsageProps = Pick<ContentProps, 'usage'>`
Not only I make sure that every props from Limit is passed on through its children, but it will also be easier to read. What do you think?